src/Illuminate/Foundation/Console/RouteListCommand.php

```diff
@@ -137,74 +138,15 @@ protected function displayRoutes(array $routes)
      */
     protected function getMiddleware($route)
     {
-        $middlewares = array_values($route->middleware());
+        $middlewares = $route->gatherMiddleware();
 
-        $actionName = $route->getActionName();
-
-        if (! empty($actionName) && $actionName !== 'Closure') {
-            $middlewares = array_merge($middlewares, $this->getControllerMiddleware($actionName));
-        }
-
-        return implode(',', $middlewares);
-    }
-
-    /**
-     * Get the middleware for the given Controller@action name.
-     *
-     * @param  string  $actionName
-     * @return array
-     */
-    protected function getControllerMiddleware($actionName)
-    {
-        $segments = explode('@', $actionName);
-
-        return $this->getControllerMiddlewareFromInstance(
-            $this->laravel->make($segments[0]),
-            isset($segments[1]) ? $segments[1] : null
-        );
-    }
-
-    /**
-     * Get the middlewares for the given controller instance and method.
-     *
-     * @param  \Illuminate\Routing\Controller  $controller
-     * @param  string|null  $method
-     * @return array
-     */
-    protected function getControllerMiddlewareFromInstance($controller, $method)
-    {
-        if (! method_exists($controller, 'getMiddleware')) {
-            return [];
-        }
-
-        $middleware = $this->router->getMiddleware();
-
-        $results = [];
-
-        foreach ($controller->getMiddleware() as $data) {
-            if (! is_string($data['middleware'])) {
-                continue;
-            }
-
-            if (! $method || ! $this->methodExcludedByOptions($method, $data['options'])) {
-                $results[] = Arr::get($middleware, $data['middleware'], $data['middleware']);
+        foreach ($middlewares as $i => $middleware) {
+            if ($middleware instanceof Closure) {
+                $middlewares[$i] = 'Closure';
             }
         }
```

Comment:
Don't know what the standards are for Laravel, but I (personally) would rewrite this with references:

```
foreach ($middlewares as &$middleware) {
    if ($middleware instanceof Closure) {
        $middleware = 'Closure';
    }
}
```